File: Selenium/Activities/Activity21.py
# Import required packages from selenium
import time

from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver using WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Set Explicit wait of 3 Secs
    wait = WebDriverWait(driver, 10)

    # Launch the URL and get the title
    driver.get("https://v1.training-support.net/selenium/tab-opener")
    home_title = driver.title
    print("Title of the Home Page is : {}".format(home_title))

    # Click on Multitab launcher Button
    tabLauncher = driver.find_element(By.ID, "launcher")
    homeWindow = driver.current_window_handle
    print("Home Window : " + homeWindow)
    tabLauncher.click()
    wait.until(expected_conditions.number_of_windows_to_be(2))

    openWindows = driver.window_handles
    print("Open Windows : " + str(openWindows))
    new_window = ""
    for openWindow in openWindows:
        if not (openWindow == homeWindow):
            driver.switch_to.window(openWindow)
            wait.until(expected_conditions.title_contains("Newtab"))

    # Print the Current Window and the title of the page
    print("Current Window is :" + driver.current_window_handle)
    print("Current Page title : " + driver.title)

    # Wait for 3 Seconds and Close the Browser
    time.sleep(3)
    driver.close()

File: Selenium/Activities/Activity21.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.Alert;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;
import java.util.List;
import java.util.Set;

public class Activity21 {
    public static void main(String[] args) throws InterruptedException {

        // Setup the Webdriver
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver= new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofMillis(1000));

        // Launch the URL and get the title
        driver.get("https://v1.training-support.net/selenium/tab-opener");
        String pageTitle = driver.getTitle();
        System.out.println(" Title of the Page is : " + pageTitle);

        // Click on Multitab launcher Button
        WebElement tabLauncher = driver.findElement(By.id("launcher"));
        String homeWindow = driver.getWindowHandle();
        System.out.println("Home Window : " + homeWindow);
        tabLauncher.click();
        wait.until(ExpectedConditions.numberOfWindowsToBe(2));

        Set<String> openWindows = driver.getWindowHandles();
        System.out.println("Open Windows : " + openWindows);
        for (String openWindow : openWindows){
            if(!(openWindow.equals(homeWindow))){
                driver.switchTo().window(openWindow);
            }
        }

        // Print the Current Window and the title of the page
        System.out.println("Current Window is :" + driver.getWindowHandle());
        System.out.println("Current Page title : " + driver.getTitle());


        // Click on Multitab launcher Button
        tabLauncher = driver.findElement(By.id("actionButton"));
        homeWindow = driver.getWindowHandle();
        System.out.println("Home Window : " + homeWindow);
        tabLauncher.click();
        wait.until(ExpectedConditions.numberOfWindowsToBe(2));

        openWindows = driver.getWindowHandles();
        System.out.println("Open Windows : " + openWindows);
        for (String openWindow : openWindows){
            if(!(openWindow.equals(homeWindow))){
                driver.switchTo().window(openWindow);
            }
        }

        // Print the Current Window and the title of the page
        System.out.println("Current Window is :" + driver.getWindowHandle());
        System.out.println("Current Page title : " + driver.getTitle());

        // Wait for 3 Seconds and close the browser
        Thread.sleep(3000);
        driver.quit();

    }
}
